xtractmime/utils.py

```diff
@@ -0,0 +1,82 @@
+from struct import unpack
+
+def _is_mp4_signature(input_bytes):
+	input_size = len(input_bytes)
+	if input_size < 12:
+		return False
+
+	box_size = unpack(">I", input_bytes[0:4])[0]
+
+	if (input_size < box_size) or (box_size % 4 != 0):
+		return False
+
+	if input_bytes[4:8] != b"ftyp":
+		return False
+
+	if input_bytes[8:11] == b"mp4":
+		return True
+
+	bytes_read = 16
+	while bytes_read < box_size:
+		if input_bytes[bytes_read:bytes_read+3] == b"mp4":
+			return True
+		bytes_read += 1
+
+	return False
+
+
+def parse_vint(input_bytes, input_size, index):
+	mask = 128
+	max_vint_size = 8
+	numbersize = 1
+
+	while numbersize < max_vint_size and numbersize < input_size:
+		if input_bytes[index] & mask != 0:
+			break
+		mask = mask >> 1
+		numbersize += 1
+
+	index = 0
+	parsednum = input_bytes[index] & ~mask
+	index += 1
+	bytes_remain = numbersize
+
+	while bytes_remain != 0:
+		parsednum = parsednum << 8
+		parsednum = parsednum | input_bytes[index]
+		index += 1
+		if index >= input_size:
+			break
+		bytes_remain -= 1
+
+	return parsednum and numbersize
```

Comment:
Since the specification does not say “Return parsed number | number size”, but “Return parsed number and number size”, I wonder if it means you should return both numbers (e.g. as a tuple). If that were the case, form what step 6.1.3 says, I would guess you would only actually need the number size. So I’m not 100% sure, but I think the right approach here may be:

```suggestion
	return numbersize
```

If I’m right here, it would also mean that you can simplify this method by removing everything about getting `parsednum`.

Note: [You are not the only one having issues with this part of the standard](https://github.com/whatwg/mimesniff/issues/93).

Reply:
I figured out that there is no need for `parsednum` only `numbersize` will be good and  `Iter >= input_size -4` this is the right condition as per this [PR](https://github.com/whatwg/mimesniff/issues/93). After these changes tests are passing for now but if something goes wrong we can change it later. ✌🏼